### PR TITLE
Calculation of Cv now allows for general value of gamma...

### PR DESCRIPTION
**Pull request by lexasov**
Calculation of Cv allows for general gamma, needed for turbulence runs with gamma=1.001 
Also changed mui to 0.62 (75% H + 23% He + 2% C mixture)

### CODE REVIEW

**Comment by sekelle** [comment]
There are build errors, see CI output. Are we using this for the runs on LUMI? Can existing checkpoints be used still?

File: sph/include/sph/positions_gpu.cu [hunk 1]
```diff
@@ -38,8 +38,8 @@ namespace sph
 template<class Tc, class Tv, class Ta, class Tm1, class Tt, class Thydro>
 __global__ void computePositionsKernel(size_t first, size_t last, double dt, double dt_m1, Tc* x, Tc* y, Tc* z, Tv* vx,
                                        Tv* vy, Tv* vz, Tm1* x_m1, Tm1* y_m1, Tm1* z_m1, Ta* ax, Ta* ay, Ta* az,
-                                       Tt* temp, Tm1* du, Tm1* du_m1, Thydro* h, Thydro* mui, Thydro constCv,
-                                       const cstone::Box<Tc> box)
+                                       Tt* temp, Tm1* du, Tm1* du_m1, Thydro* h, Thydro* mui, Thydro* gamma,
```
Comment (OsmanSeckinSimsek) [inline sph/include/sph/positions_gpu.cu]: This doesn't need to be a gamma pointer, just use the "Thydro gamma"

File: sph/include/sph/positions_gpu.cu [hunk 2]
```diff
@@ -68,29 +68,29 @@ __global__ void computePositionsKernel(size_t first, size_t last, double dt, dou
...
+    Thydro cv = (constCv < 0) ? idealGasCv(mui[i], gamma) : constCv;
     temp[i] += energyUpdate(dt, dt_m1, du[i], du_m1[i]) / cv;
     du_m1[i] = du[i];
 }
 
 template<class Tc, class Tv, class Ta, class Tm1, class Tt, class Thydro>
 void computePositionsGpu(size_t first, size_t last, double dt, double dt_m1, Tc* x, Tc* y, Tc* z, Tv* vx, Tv* vy,
                          Tv* vz, Tm1* x_m1, Tm1* y_m1, Tm1* z_m1, Ta* ax, Ta* ay, Ta* az, Tt* temp, Tm1* du, Tm1* du_m1,
-                         Thydro* h, Thydro* mui, Thydro constCv, const cstone::Box<Tc>& box)
+                         Thydro* h, Thydro* mui, Thydro* gamma, Thydro constCv, const cstone::Box<Tc>& box)
```
Comment (OsmanSeckinSimsek) [inline sph/include/sph/positions_gpu.cu]: This gamma shouldn't be a pointer as well.

File: sph/include/sph/positions_gpu.cu [hunk 3]
```diff
@@ -68,29 +68,29 @@ __global__ void computePositionsKernel(size_t first, size_t last, double dt, dou
...
-                                                      ay, az, temp, du, du_m1, h, mui, constCv, box);
+                                                      ay, az, temp, du, du_m1, h, mui, gamma, constCv, box);
 }
 
 #define POS_GPU(Tc, Tv, Ta, Tm1, Tt, Thydro)                                                                           \
     template void computePositionsGpu(size_t first, size_t last, double dt, double dt_m1, Tc* x, Tc* y, Tc* z, Tv* vx, \
                                       Tv* vy, Tv* vz, Tm1* x_m1, Tm1* y_m1, Tm1* z_m1, Ta* ax, Ta* ay, Ta* az,         \
-                                      Tt* temp, Tm1* du, Tm1* du_m1, Thydro* h, Thydro* mui, Thydro constCv,           \
-                                      const cstone::Box<Tc>& box)
+                                      Tt* temp, Tm1* du, Tm1* du_m1, Thydro* h, Thydro* mui, Thydro* gamma,            \
```
Comment (OsmanSeckinSimsek) [inline sph/include/sph/positions_gpu.cu]: no gamma pointers

**Review by OsmanSeckinSimsek — APPROVED** [review]

**Pull request merged**